jsk_footprint/src/costmap_footorint_reconfigure.cpp

```diff
@@ -0,0 +1,341 @@
+        prev_.clear();
+      }else{
+        prev_.push_back(tmp);
+      }
+    }
+    return result_vals_;
+  }
+
+  //We assumes the z value is 0
+  pcl::PointCloud<pcl::PointXYZ> parse_to_points(std::string footprint){
+    pcl::PointCloud<pcl::PointXYZ> result_points_;
+    vector<string> v;
+    vector<double> prev_;
+
+    boost::algorithm::split( v, footprint, boost::is_any_of(","));
+    BOOST_FOREACH(string str, v){
+      str.erase(boost::remove_if(str, boost::is_any_of("[] ")), str.end());
+      double tmp = atof( str.c_str() );
+      if(prev_.size() == 1){
+        pcl::PointXYZ tmp_point(prev_[0], tmp, 0);
+        result_points_.push_back(tmp_point);
+        prev_.clear();
+      }else{
+        prev_.push_back(tmp);
+      }
+    }
+    return result_points_;
+  }
+
+};
+
+class CostmapFootPrintReconfigure{
+public:
+  CostmapFootPrintReconfigure():target_cloud_(new pcl::PointCloud<pcl::PointXYZ>),n_("~"), basic_coordiname_tf_("base_footprint"){
+
+    if (!n_.getParam("consider_robotarms", consider_robotarms_))
+      {
+        ROS_WARN("~consider_robotarms is not specified");
+        consider_robotarms_ = true;
+      }
+
+    if (!n_.getParam("consider_cloud", consider_cloud_))
+      {
+        ROS_WARN("~consider_cloud is not specified");
+        consider_cloud_ = false;
+      }
+
+    if (!n_.getParam("default_footprint_string", default_footprint_string_))
+      {
+        ROS_WARN("~default_footprint_string is not specified");
+        default_footprint_string_ = std::string("[[-0.34,-0.34],[-0.44,-0.19],[-0.44,0.19],[-0.34,0.34],[0.34,0.34],[0.39,0],[0.34,-0.34]]");
+      }
+
+    if (!n_.getParam("box_offset", box_offset_))
+      {
+        ROS_WARN("~box_offset is not specified");
+        box_offset_ = 0;
+      }
+
+    if (!n_.getParam("height", height_))
+      {
+        ROS_WARN("~height is not specified");
+	height_ = 2.0;
+      }
+
+    if (!n_.getParam("frame_id", frame_id_))
+      {
+        ROS_WARN("~frame_id is not specified");
+	frame_id_ = "base_footprint";
+      }
+
+    if (!n_.getParam("enable_callrequest", enable_callrequest_))
+      {
+        ROS_WARN("~enable_callrequest is not specified");
+	enable_callrequest_ = true;
+      }
+
+    if (!n_.getParam("enable_box_publish", enable_box_publish_))
+      {
+        ROS_WARN("~enable_box_publish is not specified");
+	enable_box_publish_ = true;
+      }
+
+    //parse if consifer_robotarms is true
+    if(consider_robotarms_){
+      std::string tmp_tf_names;
+      if (!n_.getParam("tf_names", tmp_tf_names))
+        {
+          ROS_WARN("~tf_names is not specified");
+          tmp_tf_names = std::string("/larm,/rarm");
+        }
+      boost::algorithm::split( tf_names_, tmp_tf_names, boost::is_any_of(","));
+    }
+    
+    srv_ = n_.advertiseService("reconfigure_costmap_footprint", &CostmapFootPrintReconfigure::request_cloud, this);
+    sub_ = n_.subscribe("input", 1, &CostmapFootPrintReconfigure::cloud_cb,this);
```

Comment:
What is ~input pointcloud?
if the pointcloud is large like kinect/stereo pointcloud, please implement it as nodelet.